apps/user_ldap/lib/Service/BirthdateParserService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\User_LDAP\Service;
+
+use DateTimeImmutable;
+use Exception;
+use InvalidArgumentException;
+
+class BirthdateParserService {
+	/**
+	 * Try to parse the birthdate from LDAP.
+	 * Supports LDAP's generalized time syntax, YYYYMMDD and YYYY-MM-DD.
+	 *
+	 * @throws InvalidArgumentException If the format of then given date is unknown
+	 */
+	public function parseBirthdate(string $value): DateTimeImmutable {
+		if (strlen($value) >= 11) {
```

Comment:
```suggestion
		if (strlen($value) >= strlen('YYYY-MM-DD') {
```

nit: no magic numbers

Reply:
Good point. I also added a comment about the formats.